### PR TITLE
feat: abstract search list

### PR DESCRIPTION
**Pull request by shawnyama**
# Description

Abstracting search item component to accept model and dataset data. Following this design:

image
Preparing deprecation for:
- `models-listview.vue`
- `datasets-listview.vue`
- `articles-listview.vue`

Just keeping them in case someone may use their functions but I feel they are good to nuke. Would it be worth syncing with everyone to make sure?

Resolves 438 partially

### CODE REVIEW

File: packages/client/hmi-client/src/services/data.ts [hunk 1]
```diff
@@ -498,24 +498,18 @@ const fetchResource = async (
...
+					allData: await searchXDDArticles(term, searchParam?.[ResourceType.XDD]),
+					allDataFilteredWithFacets: await searchXDDArticles(
+						term,
+						searchParamWithFacetFilters?.[ResourceType.XDD]
+					)
+				});
+			} else if (resourceType === ResourceType.MODEL || resourceType === ResourceType.DATASET) {
+				resolve(
+					getAssets(term, resourceType, searchParamWithFacetFilters?.[resourceType as ResourceType])
+				);
```
Comment (YohannParis) [inline packages/client/hmi-client/src/services/data.ts:512]: Why removing the `switch`?

File: packages/client/hmi-client/src/components/data-explorer/search-results-list.vue [hunk 2]
```diff
@@ -1,44 +1,45 @@
...
+			:selected-search-items="selectedSearchItems" :search-term="searchTerm"
+			@toggle-model-selected="updateSelection" />
+		<datasets-listview v-if="resultType === ResourceType.DATASET" class="list-view"
+			:datasets="(filteredAssets as Dataset[])" :raw-concept-facets="rawConceptFacets"
+			:selected-search-items="selectedSearchItems" :search-term="searchTerm"
+			@toggle-dataset-selected="updateSelection" />
+		<articles-listview v-if="resultType === ResourceType.XDD" class="list-view"
+			:articles="(filteredAssets as XDDArticle[])" :raw-concept-facets="rawConceptFacets"
+			:selected-search-items="selectedSearchItems" @toggle-article-selected="updateSelection" />
+	</div> -->
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/data-explorer/search-results-list.vue:28]: All comments should be removed.
People can use git to see the old version.

File: packages/client/hmi-client/src/components/data-explorer/search-results-list.vue [hunk 3]
```diff
@@ -1,44 +1,45 @@
...
 
 <script setup lang="ts">
-import { computed, PropType } from 'vue';
-import ModelsListview from '@/components/data-explorer/models-listview.vue';
-import DatasetsListview from '@/components/data-explorer/datasets-listview.vue';
-import ArticlesListview from '@/components/data-explorer/articles-listview.vue';
+import { ref, computed, PropType } from 'vue';
+// import ModelsListview from '@/components/data-explorer/models-listview.vue';
+// import DatasetsListview from '@/components/data-explorer/datasets-listview.vue';
+// import ArticlesListview from '@/components/data-explorer/articles-listview.vue';
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/data-explorer/search-results-list.vue:37]: Remove comments

File: packages/client/hmi-client/src/components/data-explorer/search-results-list.vue [hunk 4]
```diff
@@ -59,70 +60,59 @@ const props = defineProps({
...
-			const docMap: { [docid: string]: XDDArticle } = {};
-			filteredXDDExtractions.value.forEach((ex) => {
-				if (ex.properties.documentBibjson === undefined) return; // skip
+// const rawConceptFacets = computed(() => {
+// 	const searchResults = props.dataItems.find((res) => res.searchSubsystem === props.resultType);
+// 	if (searchResults) {
+// 		return searchResults.rawConceptFacets;
+// 	}
+// 	return null;
+// });
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/data-explorer/search-results-list.vue:83]: Remove comments

File: packages/client/hmi-client/src/components/data-explorer/search-results-list.vue [hunk 5]
```diff
@@ -59,70 +60,59 @@ const props = defineProps({
...
-				docMap[docid].relatedExtractions?.push(ex);
-			});
-			articlesFromExtractions = Object.values(docMap) as XDDArticle[];
+			if (searchResults.xddExtractions.length > 0) {
+				const docMap: { [docid: string]: XDDArticle } = {};
+				searchResults.xddExtractions.forEach((ex) => {
+					if (ex.properties.documentBibjson === undefined) return; // skip
+
+					// eslint-disable-next-line no-underscore-dangle
+					const docid = ex.properties.documentBibjson.gddid || ex.properties.documentBibjson._gddid; // FIXME: embedded doc metadata has no proper fields mapping
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/data-explorer/search-results-list.vue:98]: This should be resolved now from @dvince2 PR.

File: packages/client/hmi-client/src/components/data-explorer/search-item.vue [hunk 6]
```diff
@@ -76,6 +202,17 @@ const formatDetails = () =>
...
+.title,
+.details,
+.description,
+.parameters,
+.features {
+	display: -webkit-box;
+	-webkit-box-orient: vertical;
+	-webkit-line-clamp: 1;
+	overflow: hidden;
+}
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/data-explorer/search-item.vue:214]: This should be moved to `src/assets/css/styles.css`

File: packages/client/hmi-client/src/components/data-explorer/search-item.vue [hunk 7]
```diff
@@ -1,17 +1,61 @@
...
 import IconAdd24 from '@carbon/icons-vue/es/add/24';
 import IconCheckmark24 from '@carbon/icons-vue/es/checkmark/24';
-import { ResultType } from '@/types/common';
+import IconOverflowMenuVertical24 from '@carbon/icons-vue/es/overflow-menu--vertical/24';
+import IconPDF16 from '@carbon/icons-vue/es/PDF/16';
+import IconLink16 from '@carbon/icons-vue/es/link/16';
+import IconChartLine16 from '@carbon/icons-vue/es/chart--line/16';
+import IconTable16 from '@carbon/icons-vue/es/table/16';
+import IconArrowLeft16 from '@carbon/icons-vue/es/arrow--left/16';
+import IconArrowRight16 from '@carbon/icons-vue/es/arrow--right/16';
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/data-explorer/search-item.vue:15]: Now that we use PrimeVue, could you use their icons:
https://www.primefaces.org/primevue/icons

File: packages/client/hmi-client/src/services/data.ts (hunk 1 above)
Comment (shawnyama) [reply packages/client/hmi-client/src/services/data.ts:512]: I wanted to use the [resourceType] as the attribute to combine the MODEL and DATASET cases. Can't write a case after case though due to the no-fallthrough eslint rule.

File: packages/client/hmi-client/src/components/data-explorer/search-item.vue [hunk 8]
```diff
@@ -4,16 +4,8 @@ import { XDDArticle, XDDExtractionType } from '@/types/XDD';
...
-import IconCheckmark24 from '@carbon/icons-vue/es/checkmark/24';
-import IconOverflowMenuVertical24 from '@carbon/icons-vue/es/overflow-menu--vertical/24';
-import IconPDF16 from '@carbon/icons-vue/es/PDF/16';
-import IconLink16 from '@carbon/icons-vue/es/link/16';
-import IconChartLine16 from '@carbon/icons-vue/es/chart--line/16';
-import IconTable16 from '@carbon/icons-vue/es/table/16';
-import IconArrowLeft16 from '@carbon/icons-vue/es/arrow--left/16';
-import IconArrowRight16 from '@carbon/icons-vue/es/arrow--right/16';
 import { ResultType, ResourceType } from '@/types/common';
+import 'primeicons/primeicons.css';
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/data-explorer/search-item.vue:8]: This is already done on the `main` branch. You don't need to add it.

**Review by YohannParis — DISMISSED** [review]

**Comment by amostafa-uncharted** [comment]
> # Description
> Abstracting search item component to accept model and dataset data. Following this design:
> 
> image Preparing deprecation for:
> 
> * `models-listview.vue`
> * `datasets-listview.vue`
> * `articles-listview.vue`
> 
> Just keeping them in case someone may use their functions but I feel they are good to nuke. Would it be worth syncing with everyone to make sure?
> 
> Resolves 438 partially

@Tom-Szendrey fyi some of the components used in the data explorer are being updated/replaced in case this may bring conflicts when you do the latest code merge

File: packages/client/hmi-client/src/components/data-explorer/search-item.vue (hunk 6 above)
Comment (shawnyama) [reply packages/client/hmi-client/src/components/data-explorer/search-item.vue:206]: Others used these class names in other components and were defined differently. Are you sure about this?

Comment (YohannParis) [reply packages/client/hmi-client/src/components/data-explorer/search-item.vue]: Sorry, I haven't seen this being classes... my bad!

**Review by YohannParis — APPROVED** [review]

**Pull request merged**